[Workers] Environments revision (#9638)
cloudflare/cloudflare-docs: content/workers/wrangler/environments.md

2. You create an environment, for example, `dev`, in the Worker's `wrangler.toml` configuration file.
3. In `wrangler.toml`, you configure the `dev` environment by [adding bindings](/workers/platform/bindings/) and/or [routes](/workers/platform/triggers/routes/).
4. You deploy the Worker using `npx wrangler deploy -e dev`.
5. In the background, Wrangler creates a new Worker named `my-worker-dev`.
6. You can now change your `my-worker` Worker code and configuration, and choose which environment to deploy your changes to.

Environments are used with the `--env` or `-e` flag on `wrangler dev`, `wrangler deploy`, and `wrangler secret`.

---

## Configuration

To configure an environment:

1. Open your Worker's `wrangler.toml` file.
2. Add `[env.<NAME>]` and change `<NAME>` to the desired name of your environment.
3. Repeat step 2 to create multiple environments.

Review the layout of an example `[env.dev]` environment that sets up a custom `dev.example.com` route:

```toml
---
filename: wrangler.toml
---
name = "your-worker"
route = "example.com"

[env.dev]
route = "dev.example.com"
```

You cannot specify multiple environments with the same name.

> Wrangler appends the environment name to the top-level name to deploy a Worker. For example, a Worker project named `my-worker` with an environment `[env.dev]` would deploy a Worker named `my-worker-dev`.

Reviewer: There are a few relevant notes here at the end of the configuration section, but after reading this part I still had the following questions:
- Do environment-specific configurations override "global" configurations in the same file? I believe this is shown in the first example, but it's not explicitly said in the text.
- Do we need to put all "global" settings before env-specific settings? If it's not required, how do we end an env-specific section? With an empty line?
- _(Related to the previous question — it only makes sense if we can add global settings anywhere on the file.)_ Is the order important, i.e., if we add environment-specific configuration first and then add global configuration settings, will the env-specific settings still apply?

Reviewer: I found relevant information about environments here: 
https://kate-fixes-env-revision.cloudflare-docs-7ou.pages.dev/workers/wrangler/configuration/#environments

We should probably link to this section, or move some of the content to the "Environments" page. What do you think?

Author: Will add a link! @pedrosousa

}
```

### Staging environment with \*.workers.dev

> In order to deploy your code to your `*.workers.dev` subdomain, include `workers_dev = true` in the desired environment. Your `wrangler.toml` file may look like this:

Reviewer: ```suggestion
To deploy your code to your `*.workers.dev` subdomain, include `workers_dev = true` in the desired environment. Your `wrangler.toml` file may look like this:
```

With this configuration, Wrangler will behave in the following manner:

```sh
~/my-worker $ npx wrangler deploy
Uploaded my-worker
Published my-worker
  example.com/*
```

```sh
~/my-worker $ npx wrangler deploy --env staging
Uploaded my-worker
Published my-worker
  https://my-worker-staging.<YOUR_SUBDOMAIN>.workers.dev
```

{{<Aside type="warning">}}

> When you create a Worker or environment, Cloudflare automatically registers an SSL certification for it. SSL certifications are discoverable and a matter of public record. Be careful when naming your Services and Environments that they do not contain sensitive information, such as, `migrating-service-from-company1-to-company2` or `company1-acquisition-load-test`.

Reviewer: Nit: Not sure if `registers an SSL certification` is clear... Do we create a new SSL/TLS certificate? Do we update an existing SSL/TLS certificate? Can we link to a page with more information on this behavior?

Also, the last sentence doesn't seem related to the first one... If that's really the case, I would create a separate paragraph for it.

With this configuration, Wrangler will behave in the following manner:

```sh
~/my-worker $ npx wrangler deploy
Uploaded my-worker
Published my-worker
  example.com/*
```

```sh
~/my-worker $ npx wrangler deploy --env staging
Uploaded my-worker
Published my-worker
  https://my-worker-staging.<YOUR_SUBDOMAIN>.workers.dev
```

{{<Aside type="warning">}}

> When you create a Worker or environment, Cloudflare automatically registers an SSL certification for it. SSL certifications are discoverable and a matter of public record. Be careful when naming your Services and Environments that they do not contain sensitive information, such as, `migrating-service-from-company1-to-company2` or `company1-acquisition-load-test`.

Author: ```suggestion
When you create a Worker via an environment, Cloudflare automatically creates an SSL certification for it. SSL certifications are discoverable and a matter of public record.
```
